contracts/QuestChain.sol

```diff
@@ -191,58 +191,86 @@ contract QuestChain is
-        external
-        override
-        onlyRole(EDITOR_ROLE)
-        whenNotPaused
-        validQuest(_questId)
-    {
-        emit QuestEdited(_msgSender(), _questId, _details);
+    function editQuest(
+        uint256[] calldata _questIdList,
+        string[] calldata _detailsList
+    ) external override onlyRole(EDITOR_ROLE) whenNotPaused {
+        uint256 _loopLength = _questIdList.length;
+
+        require(
+            _loopLength == _detailsList.length,
+            "QuestChain: invalid params"
+        );
+
+        for (uint256 i; i < _loopLength; i++) {
+            require(
+                _questIdList[i] < questCount,
+                "QuestChain: quest not found"
+            );
+        }
+
+        emit QuestEdited(_msgSender(), _questIdList, _detailsList);
     }
 
-    function submitProof(uint256 _questId, string calldata _proof)
-        external
-        override
-        whenNotPaused
-        whenQuestNotPaused(_questId)
-        validQuest(_questId)
-    {
-        Status status = _questStatus[_msgSender()][_questId];
-        require(status != Status.pass, "QuestChain: already passed");
+    function submitProof(
+        uint256[] calldata _questIdList,
+        string[] calldata _proofList
+    ) external override whenNotPaused {
+        uint256 _loopLength = _questIdList.length;
 
-        _questStatus[_msgSender()][_questId] = Status.review;
+        require(_loopLength == _proofList.length, "QuestChain: invalid params");
+
+        for (uint256 i; i < _loopLength; i++) {
+            _submitProof(_questIdList[i]);
+        }
 
-        emit QuestProofSubmitted(_msgSender(), _questId, _proof);
+        emit QuestProofSubmitted(_msgSender(), _questIdList, _proofList);
     }
 
     function reviewProof(
-        address _quester,
-        uint256 _questId,
-        bool _success,
-        string calldata _details
-    ) external override onlyRole(REVIEWER_ROLE) validQuest(_questId) {
-        Status status = _questStatus[_quester][_questId];
-        require(status == Status.review, "QuestChain: quest not in review");
-        _questStatus[_quester][_questId] = _success ? Status.pass : Status.fail;
+        address[] calldata _questorList,
+        uint256[] calldata _questIdList,
+        bool[] calldata _successList,
+        string[] calldata _detailsList
+    ) external override onlyRole(REVIEWER_ROLE) {
+        uint256 _loopLength = _questorList.length;
+
+        require(
+            _loopLength == _questIdList.length &&
+                _loopLength == _successList.length &&
+                _loopLength == _detailsList.length,
+            "QuestChain: invalid params"
+        );
+
+        for (uint256 i; i < _loopLength; i++) {
+            _reviewProof(_questorList[i], _questIdList[i], _successList[i]);
+        }
 
         emit QuestProofReviewed(
             _msgSender(),
-            _quester,
-            _questId,
-            _success,
-            _details
+            _questorList,
```

Comment:
`quester` and not `questor`

Reply:
oops, fixing.

Reply:
we used to call them questors until we found out that
> questor (plural questores) A member of the clergy who provides forgiveness for sins.
:'D